src/pages/ToDoFcUseReducer.js

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, createContext, useReducer } from "react";
+import FormFcReducer from "../components/todoFc/FormFcReducer";
+import NotesFc from "../components/todoFc/NotesFc";
+import { v4 as uuidv4 } from "uuid";
+import { useBg, getFromStorage, useLocalStorage } from "../components/customHooks";
+
+export const Context = createContext({});
+
+const initialState = {
+  tasks: getFromStorage("tasks"),
+  tasksCount: 0,
+  value: "",
+};
+
+const reducer = (state, { type, payload }) => {
+  const handlers = {
+    ADD_TASKS: () => ({
+      ...state,
+      tasks: [...state.tasks, { id: uuidv4(), title: payload, done: false }],
+      tasksCount: state.tasks.length + 1,
+    }),
+    SET_VALUE: (state, payload) => ({
+      ...state,
+      value: payload,
+    }),
+    DONE_TASK: (state, id) => ({
+      ...state,
+      tasks: state.tasks.map((task) => (task.id === id ? ((task.done = true), task) : task)),
+    }),
+    DELETE_TASK: (state, id) => ({
+      ...state,
+      tasks: state.tasks.filter((task) => task.id !== id),
+      tasksCount: state.tasksCount - 1,
+    }),
+    CLEAR_ALL: (state) => ({
+      ...state,
+      tasks: [],
+      tasksCount: 0,
+    }),
+    EDIT_TASK: (state, id) => ({
+      ...state,
+      tasks: state.tasks.filter((task) => {
+        return task.id !== id;
+      }),
+    }),
+  };
+
+  return handlers[type](state, payload) || state;
```

Comment:
what if there is no handlers[type] handler? Undefind is not a function.
(rare case in small applications but anyway)